mlflow/store/artifact/utils/models.py

```diff
@@ -63,6 +63,8 @@ def _parse_model_uri(uri):
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
     path = parsed.path
+    if parsed.fragment:
+        path = f"{path}#{parsed.fragment}"
```

Comment:
To confirm, this line will re-add the fragment to a path but separate it with a '#'?

Reply:
Yep! The `#` value delineates a path fragment in a uri path and parsing it will strip it from the model name. This just adds it back in since the input path is parsed using `urllib.parse.urlparse`